src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -1532,6 +1532,30 @@ theorem signExtend_eq_setWidth_of_lt (x : BitVec w) {v : Nat} (hv : v ≤ w):
 theorem signExtend_eq (x : BitVec w) : x.signExtend w = x := by
   rw [signExtend_eq_setWidth_of_lt _ (Nat.le_refl _), setWidth_eq]
 
+theorem toNat_signExtend_of_gt (x : BitVec w) {v : Nat} :
+  (x.signExtend (w + v)).toNat = x.toNat + (2 ^ (w + v)  - 2^w) * (if x.msb then 1 else 0) := by
```

Comment:
I see what we're going for here. I feel a nicer lemma, which is much easier to apply in practice, is:

```suggestion
theorem toNat_signExtend_of_gt (x : BitVec w) {v : Nat} (hv : v >= w):
  (x.signExtend v).toNat = x.toNat + (2^v - 2^w) * (if x.msb then 1 else 0) := by
```